src/main/java/io/r2dbc/pool/ConnectionPoolConfiguration.java

```diff
@@ -151,6 +165,10 @@ PoolMetricsRecorder getMetricsRecorder() {
 
         private PoolMetricsRecorder metricsRecorder = new SimplePoolMetricsRecorder();
 
+        private boolean registerJmx = true;
+
+        private String name = "default";
```

Comment:
Let's make this `null` and require/validate on `build()` whether a name was set. This makes it explicit to users to not register multiple JMX beans with the same name.